### PR TITLE
Bee 42975 - Automation for CreateFolder and DeleteBucketContents procedures

### PR DESCRIPTION
**Pull request by CJkrishnan**

### CODE REVIEW

File: specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        deleteProject("spec-configs-project-EC-S3")
+    }
+    def "deleteBucketContents"() {
+        when:
+        def r = deleteBucketContents.flush()
+                .config(configName)
+                .bucketName(bucketName)
+                .run()
+        then:
+        assert r.successful
```
Comment (alexey-sinyuk) [inline specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy:33]: But was it deleted? There is no assert on that.

File: specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    def createBucketContents() {
+        createBucket(bucketName)
+        def folder = pluginWithoutConfig.createFolder.flush()
+                .config(configName)
+                .bucketName(bucketName)
+                .folderName(folderName)
+                .run()
+        assert folder.isSuccessful()
```
Comment (alexey-sinyuk) [inline specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy:58]: There is no assert of creation

File: specs/src/test/groovy/com/cloudbees/pdk/hen/tests/CreateFolderSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    def 'Create folder'() {
+        when:
+        def r = createFolder.config(configName)
+                .bucketName(bucketName)
+                .folderName(folderName)
+                .run()
+        then:
+        assert r.successful
```
Comment (alexey-sinyuk) [inline specs/src/test/groovy/com/cloudbees/pdk/hen/tests/CreateFolderSpec.groovy:36]: Job is successful, but was it? No assert on folder creation

**Review by alexey-sinyuk — CHANGES_REQUESTED** [review]
Tests lack checking actual results.

File: specs/src/test/groovy/com/cloudbees/pdk/hen/tests/CreateFolderSpec.groovy (hunk 3 above)
Comment (CJkrishnan) [reply specs/src/test/groovy/com/cloudbees/pdk/hen/tests/CreateFolderSpec.groovy:36]: assertion added in last commit

File: specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy (hunk 1 above)
Comment (CJkrishnan) [reply specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy:33]: assertion added in last commit

File: specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy (hunk 2 above)
Comment (CJkrishnan) [reply specs/src/test/groovy/com/cloudbees/pdk/hen/tests/DeleteBucketContents.groovy:59]: assertion added in last commit

**Review by alexey-sinyuk — APPROVED** [review]

**Pull request merged**